libexec/bootstrap-scripts/main-deffile.sh

```diff
@@ -53,9 +53,23 @@ if [ -n "${BOOTSTRAP:-}" -a -z "${SINGULARITY_BUILDNOBASE:-}" ]; then
     fi
 fi
 
+# take a snapshot of the environment for later comparison
+if [ ${BOOTSTRAP} = "localimage" ]; then
+    export SINGULARITY_STARTING_ENVIRONMENT=$(eval_abort env -i ${SINGULARITY_libexecdir}/singularity/helpers/record-env.sh ${SINGULARITY_ROOTFS})
+    export SINGULARITY_STARTING_ENVSHA1=$(eval_abort sha1sum ${SINGULARITY_ROOTFS}/.singularity.d/env/*.sh | sha1sum)
+fi 
```

Comment:
I think we want to avoid what @gmk calls bashisms? So here first define the two variables, then export (especially because this is run as shell right?)

```
if [ ${BOOTSTRAP} = "localimage" ]; then
   SINGULARITY_STARTING_ENVIRONMENT=$(eval_abort env -i ${SINGULARITY_libexecdir}/singularity/helpers/record-env.sh ${SINGULARITY_ROOTFS})
   SINGULARITY_STARTING_ENVSHA1=$(eval_abort sha1sum ${SINGULARITY_ROOTFS}/.singularity.d/env/*.sh | sha1sum)
   export SINGULARITY_STARTING_ENVIRONMENT SINGULARITY_STARTING_ENVSHA1
fi 
```